packages/react/src/components/teleport/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { createPortal } from "react-dom"
+import { findOrCreateTeleportTarget } from "./util"
+import { ReactNode } from "react"
+
+export interface TeleportProps {
+  to: string
+  children: ReactNode
+  renderKey?: string | null | undefined
+}
+
+/**
+ * Renders child content in a portal element at the DOM root
+ * 
+ * @param to The destination portal name
+ * @param children The content to be teleported
+ * @param renderKey (optional) A key to control when the content rerenders
+ */
+export const Teleport = ({ to, children, renderKey }: TeleportProps) => {
+  const target = findOrCreateTeleportTarget(to)
+
+  return (
+    <>
+      {createPortal(children, target, renderKey)}
```

Comment:
@jake-slashid I was looking at the MUI modal docs and noticed [their warning](https://mui.com/material-ui/react-modal/#server-side-modal) about SSR and `createPortal`.

Our components should support SSR out of the box - anything we can do here to help that?

Reply:
I hadn't considered this

Reply:
We might be able to add "support" for SSR in this component behind the abstraction i.e. don't render when SSR detected or perform some alternative rendering operation 🤔

Reply:
We don't have a standard approach yet, definitely something that should be discussed. Maybe we should detect if we're server side and not use portals in that case? That could be handled on a higher level than this.

Reply:
Context in this thread: https://slashidworkspace.slack.com/archives/C04TBH6G59P/p1693311213320299

We're skipping SSR support for now